webpack/common/ForemanTasks/ForemanTasksActions.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { get } from 'foremanReact/redux/API';
+import { withInterval } from 'foremanReact/redux/middlewares/IntervalMiddleware';
+import { addToast } from 'foremanReact/redux/actions/toasts';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { foremanTaskDetailsUrl } from './ForemanTasksHelpers';
+import { foremanUrl } from '../../ForemanRhCloudHelpers';
+
+export const setupTaskPolling = ({
+  taskId,
+  key,
+  onTaskSuccess,
+  onTaskError,
+  taskErrorMessage,
+  dispatch = f => f,
+}) =>
+  withInterval(
+    get({
+      key,
+      url: foremanTaskDetailsUrl(taskId),
+      handleSuccess: ({ data }, stopTaskInterval) => {
+        if (data.result === 'success') {
+          stopTaskInterval();
+          dispatch(onTaskSuccess(data));
+        }
+        if (data.result === 'error') {
+          stopTaskInterval();
+          if (taskErrorMessage === undefined) {
+            taskErrorMessage = errorData =>
+              `${__('The task failed with the following error:')} ${
+                errorData.humanized.errors[0]
```

Comment:
Is `.humanized.errors[0]` always available? might be good to be on the safe side with `errorData?.humanized?.errors[0]`